src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -2112,6 +2112,199 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		}));
 	}
 
+	/**
+	 * Renders file changes in the input widget from timeline diff data.
+	 * This is used for agent sessions that don't have a traditional editing session
+	 * but instead have diff data computed from the timeline.
+	 * NOTE: this is pretty much an addaptation of renderChatEditingSessionState above
+	 * but using IEditSessionEntryDiff observable instead.
+	 */
+	renderFileChangesFromDiffs(diffs: IObservable<readonly IEditSessionEntryDiff[]>, sessionResource: URI) {
```

Comment:
We could merge this with the copy above, and just consolidate it based on the entries type but would like to hear from input on this approach first